rfq/order.go

```diff
@@ -175,8 +177,15 @@ func (c *AssetSalePolicy) GenerateInterceptorResponse(
 		return nil, fmt.Errorf("policy has no asset ID")
 	}
 
-	outgoingAssetAmount := uint64(htlc.AmountOutMsat / c.AskPrice)
-	htlcBalance := rfqmsg.NewAssetBalance(*c.assetID, outgoingAssetAmount)
+	// Compute the outgoing asset amount given the msat outgoing amount and
+	// the asset to BTC rate.
+	outgoingAssetAmount := rfqmath.MilliSatoshiToUnits(
+		htlc.AmountOutMsat, c.AskAssetRate,
+	)
+	amt := outgoingAssetAmount.ScaleTo(0).ToUint64()
```

Comment:
Hmm, from looking at the Godoc of `ToUint64()` the `ScaleTo(0)` _shouldn't_ be required, as it mentions "scaled down from the existing scale". But it looks like the implementation doesn't match that. Perhaps something to fix later on.